engine/src/main/java/org/terasology/engine/rendering/assets/texture/TextureData.java

```diff
@@ -70,6 +72,16 @@ public TextureData(TextureData fromCopy) {
         }
     }
 
+    /**
+     * Look up the color of a given pixel in the base mipmap level.
+     */
+    public Colorc getPixel(int x, int y) {
+        ByteBuffer baseMipmap = data[0];
+        int idx = BYTES_PER_PIXEL * (x + y * width);
+        int pixel = baseMipmap.getInt(idx);
+        return new Color(pixel);
```

Comment:
How often is this called (and how often do we believe will this be called)? 

We're creating a new `Color` with each call here. As `Color` is very similar to the JOML vectors we may borrow the same patterns from the math types. For them, we try to avoid creation of new instances, and instead rely on a destination parameter. This basically shifts the responsibility to create instances to the caller site - which hopefully knows better what they can cache or reuse.

Reply:
This is called once per block that supports tinting every time a chunk mesh is generated, so that sounds like a good idea.